ROX13642 Creating Collector Troubleshooting guide (#54763)
openshift/openshift-docs: modules/unable-to-connect-to-the-sensor.adoc

:_module-type: PROCEDURE
[id="unable-to-connect-to-the-sensor_{context}"]
= Unable to connect to the Sensor

When starting, first check if you can connect to Sensor. Sensor is responsible for downloading kernel drivers and CIDR blocks for processing network events, making it an essential part of the startup process. The following logs indicate you are unable to connect to the Sensor:

[source,terminal]
----
Collector Version: 3.12.0
OS: Ubuntu 20.04.4 LTS
Kernel Version: 5.4.0-126-generic
Starting StackRox Collector...
[INFO    2022/10/13 12:20:43] Hostname: 'hostname'
[...]
[INFO    2022/10/13 12:20:43] Sensor configured at address: sensor.stackrox.svc:9998
[INFO    2022/10/13 12:20:43] Attempting to connect to Sensor
[INFO    2022/10/13 12:21:13]
[INFO    2022/10/13 12:21:13] == Collector Startup Diagnostics: ==
[INFO    2022/10/13 12:21:13]  Connected to Sensor?       false
[INFO    2022/10/13 12:21:13]  Kernel driver available?   false
[INFO    2022/10/13 12:21:13]  Driver loaded into kernel? false
[INFO    2022/10/13 12:21:13] ====================================
[INFO    2022/10/13 12:21:13]
[FATAL   2022/10/13 12:21:13] Unable to connect to Sensor.
----

> This error could mean that the sensor has not started correctly or that the collector configuration is incorrect. To fix this issue, you should double-check the Collector configuration to ensure that the Sensor address is correct and that the Sensor pod is running correctly.

Reviewer: ```suggestion
This error could mean that the sensor has not started correctly or that the collector configuration is incorrect. To fix this issue, you must verify the Collector configuration to ensure that the Sensor address is correct and that the Sensor pod is running correctly.
```

Author: ack
